### PR TITLE
Install pre-commit

### PR DESCRIPTION
**Pull request by istride**
[Pre-commit](https://pre-commit.com/) is a framework that makes it easy to install and run git hooks. We intend to use it to ensure that [black](https://github.com/psf/black) and [flake8](https://flake8.pycqa.org/en/latest/) run on every commit.

See new dev setup instructions in 'docs/development.md'. I recommend deleting your current virtualenv and creating a new one.

### CODE REVIEW

File: src/rpft/rapidpro/models/common.py [hunk 1]
```diff
@@ -100,5 +100,5 @@ def assign_uuid(self, uuid_dict):
     def render(self):
         render_dict = {"name": self.name, "uuid": self.uuid}
         if self.query:
-            render_dict["query"] = query
+            render_dict["query"] = self.query
```
Comment (istride) [inline src/rpft/rapidpro/models/common.py:103]: @geoo89 I want to make sure this is what was originally intended. Is it a safe change?

File: src/rpft/rapidpro/models/routers.py [hunk 2]
```diff
@@ -371,8 +373,6 @@ def from_dict(data, exits):
         return RandomRouter(result_name, categories)
 
     def add_choice(self, category_name=None, destination_uuid=None):
-        if not category_name:
-            category = f"Bucket {len(self.categories) + 2}"
```
Comment (istride) [inline src/rpft/rapidpro/models/routers.py:375]: @geoo89 this was flagged as redundant; is it ok to delete?

**Comment by istride** [comment]
@edmoss345 When you have time, it would be good to get your perspective on this as a Windows user. Does this work for you?

File: src/rpft/rapidpro/models/common.py (hunk 1 above)
Comment (geoo89) [reply src/rpft/rapidpro/models/common.py:103]: Yes, this looks like a bug that you just fixed.

File: src/rpft/rapidpro/models/routers.py (hunk 2 above)
Comment (geoo89) [reply src/rpft/rapidpro/models/routers.py:375]: I think this is another bug: `category` in line 375 should be `category_name`.

**Review by geoo89 — CHANGES_REQUESTED** [review]
As mentioned, my only concern is WIP commits (so that I can change to another branch) that will be amended later, which I sometimes do. Hopefully not too big of an issue, especially if I use some IDE integration for black/isort/flake8 rather than commandline.

Only requested change is line 375 in `routers.py` (see second comment).

**Comment by istride** [comment]
@geoo89 In general, I would use [git stash](https://www.atlassian.com/git/tutorials/saving-changes/git-stash) to save work that is not yet ready to commit. Having said that, it is possible to temporarily disable commit hooks with `git commit --no-verify`, or equivalent, if you're not using the command line client.

**Comment by geoo89** [comment]
> @geoo89 In general, I would use [git stash](https://www.atlassian.com/git/tutorials/saving-changes/git-stash) to save work that is not yet ready to commit. Having said that, it is possible to temporarily disable commit hooks with `git commit --no-verify`, or equivalent, if you're not using the command line client.

Yeah, most of the time that works (and that's what I'd do), but I switch to a different branch to work on another feature, it's easy to lose track of what has been stashed. But the disable flag is definitely a good option for those cases, thanks.

**Pull request merged**